src/Tickets/Seating/Orders/Seats_Report.php

```diff
@@ -212,4 +220,33 @@ public function add_seats_row_action( $actions, $post ): array {
 
 		return $actions;
 	}
+
+	/**
+	 * Checks if the upsell should be shown in the Seats report tab instead.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	protected function should_show_upsell(): bool {
+		$seating = get_resource( 'tec-seating' );
+
+		/**
+		 * Filters whether the upsell should be shown in the Seats report tab.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool                                           $show_upsell Whether the upsell should be shown.
+		 * @param TEC\Common\StellarWP\Uplink\Resources\Resource $seating     The seating service.
+		 */
+		return apply_filters(
+			'tec_tickets_seating_should_show_upsell',
+			! (
```

Comment:
Why not use AND here?

```
$seating->get_license_object()->is_valid() &&
$seating->get_license_object()->is_expired() &&
tribe_get_option( 'tec_tickets_seating_connected_on', false )
```

Furthermore, check if the value of the option returned is actually a timestamp, else any value will do.

Reply:
It's a negation of the ORs. So i want to stop evaluation on the first true statement. I think it's faster from trying to make the expression correct but with using AND operators.

Also a license could never be valid and expired in the same time :D Unless we have a bug in uplink at least

Reply:
Now the option is a simple bool being stored, so any value should do.